src/main/java/it/pagopa/transactions/configurations/NodoConfig.java

```diff
@@ -1,42 +1,47 @@
 package it.pagopa.transactions.configurations;
 
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
 import it.pagopa.generated.nodoperpsp.model.NodoAttivaRPT;
 import it.pagopa.generated.nodoperpsp.model.NodoVerificaRPT;
 import it.pagopa.generated.transactions.model.ActivatePaymentNoticeReq;
 import it.pagopa.generated.transactions.model.VerifyPaymentNoticeReq;
 import it.pagopa.transactions.utils.NodoConnectionString;
+import lombok.extern.slf4j.Slf4j;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.ObjectMapper;
 
 @Configuration
+@Slf4j
 public class NodoConfig {
 
-    @Bean
-    public NodoConnectionString nodoConnectionString(
-                                                     @Value(
-                                                         "${nodo.connection.string}"
-                                                     ) String nodoConnectionParamsAsString
-    )
-            throws JsonProcessingException {
+    @Value("${nodo.connection.string}")
+    private String nodoConnectionParamsAsString;
 
-        return new ObjectMapper().readValue(nodoConnectionParamsAsString, NodoConnectionString.class);
-    }
+    @Autowired
+    private it.pagopa.generated.nodoperpsp.model.ObjectFactory objectFactoryNodoPerPsp;
 
-    @Bean
-    public NodoVerificaRPT baseNodoVerificaRPTRequest(
-                                                      @Value(
-                                                          "${nodo.connection.string}"
-                                                      ) String nodoConnectionParamsAsString,
-                                                      it.pagopa.generated.nodoperpsp.model.ObjectFactory objectFactoryNodoPerPsp
-    )
-            throws JsonProcessingException {
+    @Autowired
+    private it.pagopa.generated.transactions.model.ObjectFactory objectFactoryNodeForPsp;
 
-        NodoConnectionString nodoConnectionParams = nodoConnectionString(nodoConnectionParamsAsString);
+    @Bean
+    public NodoConnectionString nodoConnectionString() {
+        NodoConnectionString nodoConnectionString = null;
+        try {
+            nodoConnectionString = new ObjectMapper()
+                    .readValue(nodoConnectionParamsAsString, NodoConnectionString.class);
+        } catch (JsonProcessingException e) {
+            // exception not logged here since it can expose sensitive information contained
+            // into nodo connection string
+            log.error("Exception parsing nodo connection string");
```

Comment:
in this case we should not start the pod, and block the rolling update on the deployment on aks. in this way we block the deployment of the new app version, what do you think?

Reply:
the question is, in case of invalid connection string, springboot application starts?

Reply:
Yes, you're absolutely right, fixed and tested with wrong connections string as parameter checking that application failed to start